File: project2/Skeleton Codes/read_data.py
import pandas as pd
import numpy as np
"""
MGT - 418 
Convex Optimization - Project 2
2021-2022 Fall
Kernel Learning
"""
np.random.seed(0)

def prepare_ionosphere_dataset():
    df = pd.read_csv('ionosphere.data', sep=",", header=None)
    data_array = np.array(df)
    data_array = np.delete(data_array, 1, 1)
    labels = data_array[:, -1]
    labels[labels == 'g'] = -1
    labels[labels == 'b'] = 1
    data_array = data_array[:, :-1]
    data_normalized = data_array / data_array.max(axis=0)

    return data_normalized, labels

File: project2/Skeleton Codes/read_data.m
function [data, labels] = read_data()
A = importdata('ionosphere.data');
data = zeros(351, 34);
labels = zeros(351, 1);
for i = 1 : 351
    temp = split(A{i}, ',');
    temp_label = temp(35);
    if temp_label{1} == 'g'
        labels(i, 1) = -1;
    else
       labels(i, 1) = 1;
    end
    data_array = zeros(34, 1);
    for j = 1: 34
        data_array(j) = str2double(cell2mat(temp(j)));  
    end
%     data(i, :) = data_array;
    data(i, :) = data_array;
end
data(:, 2) = [];
data = data./ max(data);
end